### PR TITLE
SOLR-16476: Remove commons-text dependency from solr-core

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16476

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java [hunk 1]
```diff
@@ -70,20 +67,13 @@ public void doGet(HttpServletRequest _request, HttpServletResponse _response) th
         // We have to close this to flush OutputStreamWriter buffer
         out =
             new OutputStreamWriter(
-                new CloseShieldOutputStream(response.getOutputStream()), StandardCharsets.UTF_8);
+                CloseShieldOutputStream.wrap(response.getOutputStream()), StandardCharsets.UTF_8);
```
Comment (risdenk) [inline solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java:70]: Fixes deprecation warning from `CloseShieldOutputStream`

File: solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java [hunk 2]
```diff
@@ -70,20 +67,13 @@ public void doGet(HttpServletRequest _request, HttpServletResponse _response) th
...
-              StringEscapeUtils.escapeEcmaScript(request.getContextPath()),
-              StringEscapeUtils.escapeEcmaScript(CommonParams.CORES_HANDLER_PATH),
-              StringEscapeUtils.escapeEcmaScript(pack.getSpecificationVersion())
-            };
-
-        out.write(StringUtils.replaceEach(html, search, replace));
+        String html =
+            IOUtils.toString(in, StandardCharsets.UTF_8)
+                .replace("${version}", pack.getSpecificationVersion());
+        out.write(html);
```
Comment (tflobbe) [inline solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java:76]: Guessing we don't need the other replacements?

**Review by tflobbe — COMMENTED** [review]
Quick question. Will the version of `commons-text` be back at 1.9 now that we don't explicitly set it and will be transient? Is there anything we need to do to keep it at 1.10 in this case, maybe force it in `hadoop-auth`?

**Comment by risdenk** [comment]
> Will the version of commons-text be back at 1.9 now that we don't explicitly set it and will be transient? Is there anything we need to do to keep it at 1.10 in this case, maybe force it in hadoop-auth?

As of now, versions.props still sets commons-text to 1.10 so commons-text stays at 1.10. if we removed commons-text from versions.props (and versions.lock regenerated) - then commons-text would go back to 1.9.

**Review by tflobbe — APPROVED** [review]
Thanks for the clarifications!

File: solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java (hunk 2 above)
Comment (dsmiley) [reply solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java:76]: Very thorough of you -- much appreciated!

**Review by dsmiley — APPROVED** [review]

**Pull request merged**